internal/blocksync/reactor.go

```diff
@@ -517,6 +550,7 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 			// NOTE: We can probably make this more efficient, but note that calling
 			// first.Hash() doesn't verify the tx contents, so MakePartSet() is
 			// currently necessary.
+			// TODO Should we also validate against the extended commit?
```

Comment:
The light client verification doesn't rely on the extended commits, so I don't see this being necessary. Maybe I'm missing something?

Reply:
The point of that TODO is that, now, we have two commits we can use for verification:
* the commit inside `second`
* the extended commit that came along with `first`
So we could be extra-careful and validate against both, by calling `VerifyCommitLight` twice: once with each commit that we have. This is for the über-anxious like myself :-)
I would like to have @jmalicevic's input here, as she's the one really looking into this aspect.

Reply:
@sergio-mena  sure, I think it makes sense to include it as part of the verification work